themes/default/content/learn/abstraction-encapsulation/component-resources/index.md

```diff
@@ -81,14 +154,28 @@ class OurBucketComponent(pulumi.ComponentResource):
         )
         return bucket_policy
 
+bucket1 = OurBucketClass('laura-bucket-1', 'default')
+bucket1.set_policy()
 
-
+pulumi.export("bucket_name", bucket1.bucket.id)
 ```
 
-Within `super()`'s init, we pass in a name for the resource, which [we recommend]({{< relref "/docs/intro/concepts/resources/components#authoring-a-new-component-resource" >}}) being of the form `<package>:<module>:<type>` to avoid type conflicts since it's being registered alongside other resources like the Bucket resource we're calling (`aws:s3:Bucket`).
+{{% /choosable %}}
+
+With the call to `super()`, we pass in a name for the resource, which [we recommend]({{< relref "/docs/intro/concepts/resources/components#authoring-a-new-component-resource" >}}) being of the form `<package>:<module>:<type>` to avoid type conflicts since it's being registered alongside other resources like the Bucket resource we're calling (`aws:s3:Bucket`).
+
+{{% choosable language python %}}
 
 That last call in the init, `self.register_outputs({})`, passes Pulumi the expected outputs so Pulumi can read the results of the creation or update of a component resource just like any other resource, so don't forget that call! You can [register default outputs using this call]({{< relref "/docs/intro/concepts/resources/components#registering-component-outputs" >}}), as well. It's not hard to imagine we will always want the bucket name for our use case, so we pass that in as an always-given output for our component resource.
 
+{{% /choosable %}}
+
+{{% choosable language typescript %}}
```

Comment:
Side note: I had started some work on an inline choosable to allow for small paragraph changes. Maybe we want to explore that further at some point? Happy to discuss it.

Reply:
Yeah that'd definitely be handy! We should just do it. 😄

Reply:
Opened! https://github.com/pulumi/pulumi-hugo/issues/1091